docs/topics/extensions.md

```diff
@@ -115,14 +115,15 @@ fun main() {
 ## Nullable receiver
 
 Note that extensions can be defined with a nullable receiver type. These extensions can be called on an object variable
-even if its value is null, and they can check for `this == null` inside the body.
+even if it's value is null. If the receiver is `null`, then reference to `this` inside the extension function will be `null`
+as well. So you will be forced to check for `this == null` inside the body.
```

Comment:
```suggestion
even if its value is null. Since the receiver type can be `null`, then `this` can also be `null`. So when defining an extension with a nullable receiver type, it's a good idea to perform a `this == null` check inside the function body to avoid compiler errors. 
```

Reply:
I really appreciate your suggestion, but I think we can and should rethink it a bit :) 

Sentence `Since the receiver type can be null, then this can also be null` is a bit obscure. I have specifically written, that if the receiver is `null`, that `this` **_will be null_**. Suggested change is:

> can also be null

So the only difference between these two formulations is that initial version have explicitly mentioned when `this` will be `null`. IMO, it is generally better to be explicit here :)

If specify when `this` might be `null` clearly, the last suggested sentence would lose its meaning. The programmer will understand in what scenarios `this` reference will be `null` and will react accordingly. So, IMO, checking for null is not a good idea, but the necessity, which occurs under very specific circumstances :)

Reply:
I really appreciate your suggestion, but I think we can and should rethink it a bit :) 

Sentence `Since the receiver type can be null, then this can also be null` is a bit obscure. I have specifically written, that if the receiver is `null`, that `this` **_will be null_**. Suggested change is:

> can also be null

So the only difference between these two formulations is that initial version have explicitly mentioned when `this` will be `null`. IMO, it is generally better to be explicit here :)

If specify when `this` might be `null` clearly, the last suggested sentence would lose its meaning. The programmer will understand in what scenarios `this` reference will be `null` and will react accordingly. So, IMO, checking for null is not a 'good idea', but the necessity, which occurs under very specific circumstances :)

Reply:
Hi @Mikhail2048, thanks for sharing your thoughts. I've updated the suggestion to make it more explicit. The main motivation for change here from my side is to use present tense so it is in line with our style guide. 

I've also changed "it's a good idea" to the stronger "we recommend".